x-pack/plugins/security_solution/public/value_list/components/value_list_modal.tsx

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useState, useCallback } from 'react';
+import { css } from '@emotion/css';
+import {
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiFlexItem,
+  EuiFlexGroup,
+  useEuiPaddingSize,
+  EuiText,
+  EuiLoadingSpinner,
+  EuiSpacer,
+  EuiSearchBar,
+} from '@elastic/eui';
+import { useFindListItems, useGetListById } from '@kbn/securitysolution-list-hooks';
+import { FormattedDate } from '../../common/components/formatted_date';
+import { useKibana } from '../../common/lib/kibana';
+import { AddListItemPopover } from './add_list_item_popover';
+import { UploadListItem } from './upload_list_item';
+import { ListItemTable } from './list_item_table';
+import { Info } from './info';
+import type { SortFields, ValueListModalProps, OnTableChange, Sorting } from '../types';
+import {
+  INFO_UPDATED_AT,
+  INFO_TYPE,
+  INFO_UPDATED_BY,
+  INFO_TOTAL_ITEMS,
+  getInfoTotalItems,
+} from '../translations';
+
+const modalWindow = css`
+  min-height: 90vh;
+  margin-top: 5vh;
+  max-width: 1400px;
+  min-width: 700px;
+`;
+
+const tableStyle = css`
+  overflow: scroll;
+`;
+
+export const ValueListModal = ({ listId, onCloseModal, canWriteIndex }: ValueListModalProps) => {
+  const [filter, setFilter] = useState('');
+  const http = useKibana().services.http;
+  const [pageIndex, setPageIndex] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
+  const [sortField, setSortField] = useState<SortFields>('updated_at');
+  const [sortOrder, setSortOrder] = useState<'asc' | 'desc'>('desc');
+
+  const {
+    data: listItems,
+    isLoading: isListItemsLoading,
+    isError,
+  } = useFindListItems({
+    listId,
+    pageIndex: pageIndex + 1,
+    pageSize,
+    sortField,
+    sortOrder,
+    filter,
+    http,
+  });
+
+  const { data: list, isLoading: isListLoading } = useGetListById({ http, id: listId });
+
+  const onTableChange: OnTableChange = ({ page, sort }) => {
+    if (page) {
+      setPageIndex(page.index);
+      setPageSize(page.size);
+    }
+    if (sort) {
+      setSortField(sort.field as SortFields);
+      setSortOrder(sort.direction);
+    }
+  };
+
+  const modalBodyStyle = css`
+    overflow: hidden;
+    padding: ${useEuiPaddingSize('m')};
+  `;
+
+  const sorting: Sorting = {
+    sort: {
+      field: sortField,
+      direction: sortOrder,
+    },
+    enableAllColumns: false,
+  };
+
+  const pagination = {
+    pageIndex,
+    pageSize,
+    totalItemCount: listItems?.total ?? 0,
+    pageSizeOptions: [5, 10, 25],
+  };
+
+  const onQueryChange = useCallback((params) => {
+    setFilter(params.queryText);
+  }, []);
+
+  const isListExist = !(isListLoading || !list);
```

Comment:
De Morgan says: reduce your expressions!

```suggestion
  const isListExist = !isListLoading && !!list;
```